ddtrace/utils/reraise.py

```diff
@@ -0,0 +1,5 @@
+def _reraise(tp, value, tb=None):
+    """Python 2 re-raise function. This function is internal and
+    will be replaced entirely with the `six` library.
+    """
+    raise tp, value, tb
```

Comment:
Due to my latest comment on #472, I'm concerned that this won't fix the issue (but I still can't replicate it consistently, so I can't prove it). I think we don't even need this new file. I would just define this within the `if PY2:` block in `compat.py` above. I'm concerned about even defining this function unless `if PY2` is `True`.

Reply:
I think the syntax error will manifest if it's read by the interpreter while preparing the module. Because of that, importing only if `PY2` should do the trick.

Anyway, I've added a test for that part, but I can't run it in Pylons because it's not Python 3 compatible.

Reply:
Well, my concern was kind of two-fold. Above, you're doing (pseudocode) "if Python 2 then import this other, new bit of code from the same project, else define this function with the same name." The problem with that is that the other, new bit of code you've added (this file) could easily be imported directly (without using `compat`) but some other developer somewhere else in the project, and then we'd have the same problem again. It shouldn't be possible to define this version of this method in Python 3.

However, like you said in the PR comments, this is probably temporary until it either gets cleaned up or replaced by Six, so I'm not going to get hung up on it. I just prefer for the code to be clean in the first place, because I know (from experience) that tech debt rarely gets paid off.

Reply:
Exactly. The change to `six` requires way more work, and I don't want to make this huge change for one fix. I'd rather prefer to solve the issue and then start the migration. Currently we have OpenTracing support as outstanding task, after that we'll proceed with the release to avoid that technical debt.